src/bin/pg_dump/pg_dump.c

```diff
@@ -2155,9 +2157,25 @@ dumpTableData_insert(Archive *fout, const void *dcontext)
 			appendPQExpBufferStr(q, "NULL");
 		else
 			appendPQExpBufferStr(q, fmtId(tbinfo->attnames[i]));
+		if (pg_strcasecmp(tbinfo->attnames[i], "orig_loginname") == 0)
```

Comment:
I think this will check column names for all tables. Could you move this out of the for-loop so that we can save some unnecessary string comparisons? For example,
```
bool is_babelfish_authid_login_ext = (isBabelfishDatabase(fout) && tbinfo->dobj.namespace &&
		pg_strcasecmp(tbinfo->dobj.namespace->dobj.name, "sys") == 0 &&
		pg_strcasecmp(tbinfo->dobj.name, "babelfish_authid_login_ext") == 0);

if (is_babelfish_authid_login_ext)
{
    for (i = 0; i < tbinfo->numatts; i++)
    {
        if (pg_strcasecmp(tbinfo->attnames[i], "orig_loginname") == 0)
            found = true
    }
}
```